### PR TITLE
Add list-secrets CLI

### PR DESCRIPTION
**Pull request by wallyworld**
Add the list-secrets CLI.
This includes:
- CLI
- api client
- api facade backend
- state list secrets

## QA steps

bootstrap and deploy a unit, eg ubuntu
create a secret
`juju exec --unit ubuntu/0 "secret-create mysecret foo=bar`

list the secret
`juju secrets`
`juju secrets --format yaml`
`juju secrets --format yaml --show-secrets`

### CODE REVIEW

**Comment by wallyworld** [comment]
[list-secrets CLI](https://trello.com/c/eXcLBTxO/1721-list-secrets-cli)

File: apiserver/facades/client/secrets/secrets.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+		if err := s.checkCanRead(); err != nil {
+			return result, errors.Trace(err)
+		}
+	}
+	ctx := context.Background()
+	metadata, err := s.secretsService.ListSecrets(ctx, secrets.Filter{})
+	if err != nil {
+		return result, errors.Trace(err)
+	}
+	for i, m := range metadata {
```
Comment (hpidcock) [inline apiserver/facades/client/secrets/secrets.go:92]: preallocate result.Results slice

File: apiserver/facades/client/secrets/secrets.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+				Path:           m.Path,
+			}
+			val, err := s.secretsService.GetSecretValue(ctx, URL)
+			valueResult := &params.SecretValueResult{
+				Error: apiservererrors.ServerError(err),
+			}
+			if err == nil {
+				valueResult.Data = val.EncodedValues()
+			}
+			result.Results[i].Value = valueResult
```
Comment (hpidcock) [inline apiserver/facades/client/secrets/secrets.go:119]: refactor to remove the index lookup e.g:
```
		secretResult := params.ListSecretResult{
			Path:        m.Path,
			Scope:       string(m.Scope),
			Version:     m.Version,
			Description: m.Description,
			Tags:        m.Tags,
			ID:          m.ID,
			ProviderID:  m.ProviderID,
			Revision:    m.Revision,
			CreateTime:  m.CreateTime,
			UpdateTime:  m.UpdateTime,
		}
		if arg.ShowSecrets {
			// ...
			secretResult.Value = valueResult
		}
		result.Results = append(result.Results, secretResult)
```

File: cmd/juju/commands/main.go [hunk 3]
```diff
@@ -597,6 +598,11 @@ func registerCommands(r commandRegistry) {
 	r.Register(charmhub.NewFindCommand())
 	r.Register(charmhub.NewDownloadCommand())
 
+	// Secrets.
+	if featureflag.Enabled(feature.Secrets) {
+		r.Register(secrets.NewListSecretsCommand())
```
Comment (hpidcock) [inline cmd/juju/commands/main.go:603]: :heavy_check_mark: nice

File: cmd/juju/secrets/list.go [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+	secrets, ok := value.([]secretDisplayDetails)
+	if !ok {
+		return errors.Errorf("expected value of type %T, got %T", secrets, value)
+	}
+
+	tw := output.TabWriter(writer)
+	w := output.Wrapper{tw}
+	w.SetColumnAlignRight(1)
+
+	w.Println("ID", "Scope", "Revision", "Path")
```
Comment (hpidcock) [inline cmd/juju/secrets/list.go:168]: Does path show which backend is used? Tabular should probably print out that this model's secrets are stored in the controller vs HC vault

Comment (hpidcock) [reply cmd/juju/secrets/list.go:168]: Also tabular should show an age field/last changed/last accessed. Important for auditing.

e.g. 
ID | Scope | Revision | Path | Age
1 bla 12 secret://... 5d

**Review by hpidcock — APPROVED** [review]
I'm happy for this to land since its just the listing command. Create/update commands are probably going to be more interesting.

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**